packages/cli/test/integration/lib/check-loopback-project.suite.js

```diff
@@ -9,6 +9,44 @@ const sinon = require('sinon');
 const chalk = require('chalk');
 const testUtils = require('../../test-utils');
 const fs = require('mem-fs-editor').create(require('mem-fs').create());
+const pkgJson = require('../../../package.json');
+
+/**
+ * Get current versions for `@loopback/context` and `@loopback/core` from
+ * the template
+ */
+const contextVer = pkgJson.config.templateDependencies['@loopback/context'];
+const coreVer = pkgJson.config.templateDependencies['@loopback/core'];
+const buildVer = pkgJson.config.templateDependencies['@loopback/build'];
+
+/***
+ * Parse version range (for example, `'^1.24.3'`) to an object such as
+ * `{major: 1, minor: 24, patch: 3}`
+ *
+ * @param {string} versionRange
+ */
+function parseVersion(versionRange) {
+  const result = /(\^|\~)(\d+)\.(\d+)\.(\d+)/.exec(versionRange);
+  return {major: +result[2], minor: +result[3], patch: +result[4]};
```

Comment:
Have you considered using `semver.parse`?

Quoting from https://www.npmjs.com/package/semver#functions:

> `parse(v)`: Attempt to parse a string as a semantic version, returning either a `SemVer` object or `null`.

Reply:
Unfortunately `parse` does not allow a version range such as `^1.0.0`.